### PR TITLE
CI: add sanitizers

### PR DESCRIPTION
**Pull request by davidbolvansky**
Enable undefined behaviour sanitizer and address sanitizer.

### CODE REVIEW

**Comment by apinski-cavium** [comment]
Note GCC has a similar (same) sanitizers that can be run there too.
I found GCC's address sanitizer finding more use out of scope issues than clang's does find.

Also I noticed you had to disable the leak detection. I could find you can only disable it at runtime via the `ASAN_OPTIONS=detect_leaks=0 ` env variable settings.

**Comment by davidbolvansky** [comment]
And clang somehow crashes in CI.
So yes, I will switch to gcc build.

**Comment by davidbolvansky** [comment]
Ready for review.

**Comment by fruffy** [comment]
Do you want this as a nightly test by any chance? Or should we merge the sanitizer test with the normal test runner?

**Comment by davidbolvansky** [comment]
As the normal one.

**Comment by fruffy** [comment]
Then let's merge `build-linux` and `build-linux-sanitizers`.  `build-linux-sanitizers` seems like a superset to the other test to me and the additional sanitizer checks do not increase the compile time noticeably.

**Comment by fruffy** [comment]
Are we actually looking at the “maybe-uninitialized” warnings? If not, I would go with the second option. 

Otherwise, we can keep the duplicate CI run. I am guessing we have to run the tests to trigger some sanitizer assertions, right?

**Comment by davidbolvansky** [comment]
>> Are we actually looking at the “maybe-uninitialized” warnings? If not, I would go with the second option.

I look on them and they are highly unlikely to happen in practice. What is worse, sometimes innocent code change triggers for example gcc inliner and then gcc emits (from middle-ene!) warnings randomly..

Second option sounds fine to me, so I will update this patch. 

And yes to the last question.

**Review by fruffy — APPROVED** [review]

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -32,9 +32,9 @@ jobs:
         key: test-${{ matrix.unified }}-${{ runner.os }}-gcc
         max-size: 1000M
 
-    - name: Build (Ubuntu Linux, GCC)
+    - name: Build (Ubuntu Linux, GCC, Sanitizers)
```
Comment (fruffy) [inline .github/workflows/ci-test.yml:35]: Just a nit, maybe also rename the top-level test to include this name. Then the label immediately shows in the action panel.

**Comment by davidbolvansky** [comment]
Feel free to merge it if you are fine with the current version of this PR, @fruffy

**Comment by davidbolvansky** [comment]
Thanks for the review and feedback, @fruffy

**Pull request merged**